crates/web5/src/dids/bearer_did.rs

```diff
@@ -67,24 +40,183 @@ impl BearerDid {
+    use crate::{test_helpers::UnitTestSuite, test_name};
+    use lazy_static::lazy_static;
+
+    mod from_portable_did {
+        use super::*;
+
+        lazy_static! {
+            static ref TEST_SUITE: UnitTestSuite =
+                UnitTestSuite::new("bearer_did_from_portable_did");
+        }
+
+        #[test]
+        fn z_assert_all_suite_cases_covered() {
+            // fn name prefixed with `z_*` b/c rust test harness executes in alphabetical order,
+            // unless intentionally executed with "shuffle" https://doc.rust-lang.org/rustc/tests/index.html#--shuffle
+            // this may not work if shuffled or if test list grows to the extent of 100ms being insufficient wait time
+
+            // wait 100ms to be last-in-queue of mutex lock
+            std::thread::sleep(std::time::Duration::from_millis(100));
+
+            TEST_SUITE.assert_coverage()
+        }
+
+        #[test]
+        fn test_can_instantiate_successfully() {
+            TEST_SUITE.include(test_name!());
+
+            let key_manager = Arc::new(InMemoryKeyManager::new());
+            let did_jwk_bearer_did = DidJwk::create(Some(DidJwkCreateOptions {
+                key_manager: Some(key_manager.clone()),
+                ..Default::default()
+            }))
+            .unwrap();
+
+            let portable_did = PortableDid {
+                did_uri: did_jwk_bearer_did.did.uri,
+                document: did_jwk_bearer_did.document,
+                private_jwks: key_manager.export_private_jwks().unwrap(),
+            };
+
+            let result = BearerDid::from_portable_did(portable_did);
+            assert!(result.is_ok());
+        }
+    }
+
+    mod get_signer {
+        use super::*;
 
-    #[test]
-    fn can_instantiate_did_jwk() {
-        let did_uri = "did:jwk:eyJrdHkiOiJPS1AiLCJ1c2UiOiJzaWciLCJjcnYiOiJFZDI1NTE5Iiwia2lkIjoiVnRTSFhQbEtEdzFFRW9PajVYTjNYV2hqU1BZVk52WC1lNHZqUk8weVlKQSIsIngiOiJpejcwc3ZTTHhOWmhzRHhlSlFfam5PVmJYM0tGTmtjQmNNaldqWm1YRXNBIiwiYWxnIjoiRWREU0EifQ";
-        let key_manager = InMemoryKeyManager::new();
+        lazy_static! {
+            static ref TEST_SUITE: UnitTestSuite = UnitTestSuite::new("bearer_did_get_signer");
+        }
+
+        #[test]
+        fn z_assert_all_suite_cases_covered() {
+            // fn name prefixed with `z_*` b/c rust test harness executes in alphabetical order,
+            // unless intentionally executed with "shuffle" https://doc.rust-lang.org/rustc/tests/index.html#--shuffle
+            // this may not work if shuffled or if test list grows to the extent of 100ms being insufficient wait time
+
+            // wait 100ms to be last-in-queue of mutex lock
+            std::thread::sleep(std::time::Duration::from_millis(100));
+
+            TEST_SUITE.assert_coverage()
+        }
+
+        #[test]
+        fn test_query_requirements_not_satisfied() {
+            TEST_SUITE.include(test_name!());
+
+            let bearer_did = DidJwk::create(None).unwrap();
```

Comment:
thoughts on create_default()?

```
    pub fn create(options: Option<Options>) -> Self {
        // Your logic here using `options`
    }

    pub fn create_default() -> Self {
        Self::create(None)
    }
    ```

Reply:
and then in kotlin we can overload actually have create() and create(options)

Reply:
I agree but would prefer to keep the APID surface area as minimal as possible for the time being

This is a good case study of, if we weren't targeting many languages all with the same API surface, then we could leverage an idiom unique to rust and implement the `Default` trait for the `DidJwk` struct, but I digress

Reply:
in other words, the wiser strategy is to accept a bit more verbosity in the name of supporting many languages, and fine tune it over time